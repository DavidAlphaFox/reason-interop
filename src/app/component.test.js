import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from './component';

describe('App component', () => {
  it('renders stories', () => {
    const stories = [{ id: 1, title: 'New phones', url: 'apple.com' }];
    const getTopStories = () => {};

    const result = shallow(
      <App stories={stories} loaded getTopStories={getTopStories} />,
    );

    expect(toJson(result)).toMatchSnapshot();
  });

  it('renders loading message when not loaded', () => {
    const getTopStories = () => {};

    const result = shallow(
      <App stories={[]} loaded={false} getTopStories={getTopStories} />,
    );

    expect(toJson(result)).toMatchSnapshot();
  });


  it('renders stories', () => {
    const stories = [
      { id: 1, title: 'New phones', url: 'apple.com' },
      { id: 2, title: 'I hate everything', url: 'medium.com' },
    ];
    const getTopStories = () => {};

    const result = shallow(
      <App stories={stories} loaded getTopStories={getTopStories} />,
    );

    expect(result.find('ul').children().length).toEqual(stories.length);
  });

  it('fetches dependancies on mount', () => {
    const stories = [];
    const getTopStories = jest.fn();

    shallow(
      <App stories={stories} loaded getTopStories={getTopStories} />,
      { lifecycleExperimental: true },
    );

    expect(getTopStories).toHaveBeenCalled();
  });
});
