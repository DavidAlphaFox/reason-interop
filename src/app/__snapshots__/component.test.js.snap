// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders loading message when not loaded 1`] = `
<div>
  <Header>
    Reason React Hacker News
  </Header>
  Loading...
</div>
`;

exports[`App component renders stories 1`] = `
<div>
  <Header>
    Reason React Hacker News
  </Header>
  <ul>
    <li>
      <a
        href="apple.com"
      >
        New phones
      </a>
    </li>
  </ul>
</div>
`;
